src/sql/workbench/api/node/mainThreadConnectionManagement.ts

```diff
@@ -117,6 +121,30 @@ export class MainThreadConnectionManagement implements MainThreadConnectionManag
 		return connection;
 	}
 
+	private convertToConnectionProfile(profile: IConnectionProfile): azdata.connection.ConnectionProfile {
+		if (!profile) {
+			return undefined;
+		}
+
+		profile = this._connectionManagementService.removeConnectionProfileCredentials(profile);
+		let connection: azdata.connection.ConnectionProfile = {
+			providerId: profile.providerName,
+			connectionId: profile.id,
+			options: profile.options,
```

Comment:
Should we clone the options here? This way, if someone edits this it'll change for both.

Reply:
What Kevin said - you should be doing a clone here otherwise updates to the options of one will also update the other. 

---
In reply to: [293955300](https://github.com/microsoft/azuredatastudio/pull/5651#discussion_r293955300) [](ancestors = 293955300)